doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_name_mappings_interceptor.md

```diff
@@ -0,0 +1,181 @@
+
+* The FIWARE service Http header sent with the notification.
+* The FIWARE service path Http header sent with the notification.
+* Any entity ID within the notification.
+* Any entity type within the notification.
+* Any attribute name within the notification.
+* Any attribute type within the notification.
+
+In fact, there is no replacement but another twin notification with propper mapped names is created. Then, both notifications are put into the destination channels within a `NGSIEvent` object:
+
+* Original headers in the Flume `Event` object are extended with mapped versions of FIWARE service and FIWARE service path.
+* An already parsed `NotifyContextRequest` object contains the original notification.
+* An already parsed `NotifyContextRequest` object contains the mapped notification.
+
+[Top](#top)
+
+###<a name="section1.1"></a>Name mappings syntax
+There exists a <i>name mappings</i> file containing a Json following this format:
+
+```
+{
+   "serviceMappings": [
+      {
+         "originalService": "<original_service>",
+         "newService": "<new_service>",
+         "servicePathMappings": [
+            {
+               "originalServicePath": "<original_service_path>",
+               "newServicePath": "<new_service_path>",
+               "entityMappings": [
+                  {
+                     "originalEntityId": "<original_entity_id>",
+                     "originalEntityType": "<original_entity_type>",
+                     "newEntityId": "<new_entity_id>",
+                     "newEntityType": "<new_entity_type>",
+                     "attributeMappings": [
+                        {
+                           "originalAttributeName": "<original_attribute_name>",
+                           "originalAttributeType": "<original_attribute_type>",
+                           "newAttributeName": "<new_attribute_name>",
+                           "newAttributeType": "<new_attribute_type>"
+                        },
+                        ...
+                     ]
+                  },
+                  ...
+               ]
+            },
+            ...
+         ]
+      },
+      ...
+   ]
+}
+```
+
+The above Json is quite straightforward: the `<original_service>` is mapped as the `<new_service>`, the `<original_service_path>` is mapped as the `<new_service_path>` and so on. The name mappings are iterated until a map is found; if no map is found, the mapped vesion of the original notification is equals to the original one.
+
+[Top](#top)
+
+###<a name="section1.2"></a>Headers before and after intercepting
+Before interception, these are the headers added by the [`NGSIRestHandler`](./ngsi_rest_handler.md) to all the internal Flumm events of type `Event`:
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1198/commits/3f5e44ab8c66588e5fa37a684e128ebd50852f79